scene/2d/cpu_particles_2d.cpp

```diff
@@ -461,14 +462,17 @@ void CPUParticles2D::set_emission_shape(EmissionShape p_shape) {
 	ERR_FAIL_INDEX(p_shape, EMISSION_SHAPE_MAX);
 	emission_shape = p_shape;
 	notify_property_list_changed();
+	queue_redraw();
 }
 
 void CPUParticles2D::set_emission_sphere_radius(real_t p_radius) {
 	emission_sphere_radius = p_radius;
+	queue_redraw();
```

Comment:
This should be wrapped in TOOLS_ENABLED and `is_editor_hint()`.
Also non-trivial setters need value guards:
```C++
if (p_radius == emission_sphere_radius) {
	return;
}
```
etc.

Reply:
Done